lib/src/iterables/partition.dart

```diff
@@ -15,35 +15,35 @@
 part of quiver.iterables;
 
 /// Partitions the input iterable into lists of the specified size.
-Iterable<List> partition(Iterable iterable, int size) {
-  return iterable.isEmpty ? [] : new _Partition(iterable, size);
+Iterable<List<T>> partition<T>(Iterable<T> iterable, int size) {
+  return iterable.isEmpty ? [] : new _Partition<T>(iterable, size);
 }
 
-class _Partition extends IterableBase<List> {
-  final Iterable _iterable;
+class _Partition<T> extends IterableBase<List<T>> {
+  final Iterable<T> _iterable;
   final int _size;
 
   _Partition(this._iterable, this._size) {
     if (_size <= 0) throw new ArgumentError(_size);
   }
 
-  Iterator<List> get iterator =>
-      new _PartitionIterator(_iterable.iterator, _size);
+  Iterator<List<T>> get iterator =>
+      new _PartitionIterator<T>(_iterable.iterator, _size);
 }
 
-class _PartitionIterator implements Iterator<List> {
-  final Iterator _iterator;
+class _PartitionIterator<T> implements Iterator<List<T>> {
+  final Iterator<T> _iterator;
   final int _size;
-  List _current;
+  List<T> _current;
 
   _PartitionIterator(this._iterator, this._size);
 
   @override
-  List get current => _current;
+  List<T> get current => _current;
 
   @override
   bool moveNext() {
-    var newValue = [];
+    var newValue = <T>[];
     var count = 0;
     for (; count < _size && _iterator.moveNext(); count++) {
```

Comment:
Could you please also change this `for` loop to a `while` loop while you're here? It's unreadable.

Reply:
Done!